rasa/core/validator.py

```diff
@@ -187,13 +190,41 @@ def verify_utterances_in_stories(self, ignore_warnings: bool = True) -> bool:
 
         return everything_is_alright
 
+    def verify_story_structure(
+        self, ignore_warnings: bool = True, max_history: int = 5
+    ) -> bool:
+        """Verifies that bot behaviour in stories is deterministic."""
+
+        logger.info("Story structure validation...")
+        logger.info(
+            f"Considering the preceding {max_history} turns for conflict analysis."
+        )
+
+        trackers = TrainingDataGenerator(
+            self.story_graph,
+            domain=self.domain,
+            remove_duplicates=False,
+            augmentation_factor=0,
+        ).generate()
+
+        # Create a list of `StoryConflict` objects
+        conflicts = find_story_conflicts(trackers, self.domain, max_history)
+
+        if len(conflicts) == 0:
```

Comment:
optional: a bit shorter and probably also avoids counting everything
```suggestion
        if not conflicts:
```